specs/Popup-AdditionalLayoutProperties-Spec.md

```diff
@@ -0,0 +1,218 @@
+    PopupPlacementMode ActualPlacement { get; }
+    
+    public event System.EventHandler<object> PlacementChanged;
+}
+```
+
+The example below shows how the `Placement` APIs are used by the 
+[CommandBarFlyout](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.CommandBarFlyout)
+to position its
+[CommandBarFlyoutCommandBar](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.CommandBarFlyoutCommandBar)'s
+secondary commands Popup, and how to respond to the event raised when XAML places the Popup.
+
+```xml
+<!-- Part of the CommandBarFlyoutCommandBar's default template -->
+<Popup
+    x:Name="OverflowPopup"
+    PlacementTarget="{Binding ElementName=PrimaryItemsRoot}"
+    DesiredPlacement="Bottom">
+</Popup>
+```
+
+```csharp
+void OnApplyTemplate()
+{
+    m_overflowPopup = GetTemplateChild("OverflowPopup");
+    m_overflowPopup.PlacementChanged += OnOverflowPopupPlacementChanged;
+}
+
+void OnOverflowPopupPlacementChanged(object sender, object args)
+{
+    UpdateVisualState(useTransitions: false);
+}
+
+void UpdateVisualState(bool useTransitions)
+{
+    if (m_overflowPopup.ActualPlacement == PopupPlacementMode.Top)
+    {
+        VisualStateManager.GoToState(this, "ExpandedUp", useTransitions);
+    }
+    else
+    {
+        VisualStateManager.GoToState(this, "ExpandedDown", useTransitions);
+    }
+}
+```
+
+## Popup.PlacementTarget property
+
+Use this property to describe which element the `Popup` should be positioned relative to.
+Defaults to `null`.
+
+If this is `null`, then `DesiredPlacement` is ignored, `ActualPlacement` is always `None`, and
+`PlacementChanged` is never raised.  If the `Popup` is in the visual tree, `PlacementTarget` will override what its
+position would otherwise be set to by layout.  Setting `PlacementTarget` to an element under a different XAML root than
+`Popup.XamlRoot` is invalid and will throw an `ArgumentException`.
+
+_Spec note: this property is analogous to the
+[FlyoutBase.Target](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Primitives.FlyoutBase.Target)
+property, but `Popup.Target` looked confusing, so we added the `Placement` prefix._
+
+## Popup.DesiredPlacement property
+
+Use this property to describe how the you would ideally like the `Popup`
+positioned relative to `PlacementTarget`.  Defaults to `None`.
+
+If this is `None`, then `PlacementTarget` is ignored,
+`ActualPlacement` is always `None` and `PlacementChanged` is never raised. 
+If both `DesiredPlacement` and `PlacementTarget` are set and `HorizontalOffset` and/or `VerticalOffset`
+are also set, then the latter two properties will offset the `Popup` from where it would have been
+placed by `DesiredPlacement` and `PlacementTarget` alone.
+
+## Popup.ActualPlacement property
+
+Use this read-only property to determine where the popup was positioned.
+Will always be `None` if either `PlacementTarget` and `DesiredPlacement` are not set.
```

Comment:
ActualPlacement doesn't really have any meaning when the popup isn't being displayed.  The scenario for PlacementChanged is to update your visual state in preparation for the popup being drawn to the screen, so it should be fine to just leave ActualPlacement with whatever value it previously had until the popup is displayed again.

Reply:
I'd say might as well reset it to None, but then there's a question of whether we should raise the event. To avoid that, I'd just say leave it at whatever it was and document it. You shouldn't use the Changed event as a way to detect that the popup has closed.

Reply:
Sounds good.